VotingApplication/VotingApplication.Web.Api.Tests/E2E/BasicPollTests.cs

```diff
@@ -0,0 +1,603 @@
+        [TestClass]
+        public class NamedVotersPollConfiguration
+        {
+            private static IVotingContext _context;
+
+            private static Poll _namedBasicPoll;
+
+            private IWebDriver _driver;
+
+            [TestInitialize]
+            public virtual void TestInitialise()
+            {
+                _driver = new NgWebDriver(new ChromeDriver(ChromeDriverDir));
+                _driver.Manage().Timeouts().SetScriptTimeout(TimeSpan.FromSeconds(10));
+                _driver.Manage().Timeouts().SetPageLoadTimeout(TimeSpan.FromSeconds(10));
+            }
+
+            [TestCleanup]
+            public void TestCleanUp()
+            {
+                _driver.Dispose();
+            }
+
+            [ClassInitialize]
+            public static void ClassInitialise(TestContext testContext)
+            {
+                _context = new TestVotingContext();
+
+                List<Option> testPollOptions = new List<Option>() {
+                new Option(){ Name = "Test Option 1", Description = "Test Description 1" },
+                new Option(){ Name = "Test Option 2", Description = "Test Description 2" }};
+
+                // Open, Named voters, No Option Adding, Shown Results
+                _namedBasicPoll = new Poll()
+                {
+                    UUID = Guid.NewGuid(),
+                    PollType = PollType.Basic,
+                    Name = "Test Poll",
+                    LastUpdated = DateTime.Now,
+                    CreatedDate = DateTime.Now,
+                    Options = testPollOptions,
+                    InviteOnly = false,
+                    NamedVoting = true,
+                    OptionAdding = false,
+                    HiddenResults = false
+                };
+
+                _context.Polls.Add(_namedBasicPoll);
+                _context.SaveChanges();
+            }
+
+            [ClassCleanup]
+            public static void ClassCleanup()
+            {
+                PollClearer pollTearDown = new PollClearer(_context);
+                pollTearDown.ClearPoll(_namedBasicPoll);
+
+                _context.Dispose();
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_VoteWithNoName_PromptsForName()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _namedBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                buttons.First(b => b.Text == "Vote").Click();
+
+                Assert.AreEqual(SiteBaseUri + "Poll/#/Vote/" + _namedBasicPoll.UUID, _driver.Url);
+
+                IWebElement formName = _driver.FindElement(NgBy.Model("loginForm.name"));
+                Assert.IsTrue(formName.IsVisible());
+                Assert.AreEqual(String.Empty, formName.Text);
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_NameInput_AcceptsValidName()
```

Comment:
I meant more that it's checking that the Go button is enabled, rather than any validation going on. Although I guess they're kind of the same thing, as it needs to be a valid name to enable it.